modules/core/src/v2/coins/abstractUtxoCoin.ts

```diff
@@ -434,14 +435,26 @@ export abstract class AbstractUtxoCoin extends BaseCoin {
    * @param actualOutputs
    * @returns {Array}
    */
-  protected static findMissingOutputs(expectedOutputs: Output[], actualOutputs: Output[]): Output[] {
+  protected static findMissingOutputs(wallet: Wallet, expectedOutputs: Output[], actualOutputs: Output[]): Output[] {
     const keyFunc = ({ address, amount }: Output): string => `${address}:${Number(amount)}`;
     const groupedOutputs = _.groupBy(expectedOutputs, keyFunc);
 
     actualOutputs.forEach((output) => {
       const group = groupedOutputs[keyFunc(output)];
       if (group) {
         group.pop();
+      } else if (wallet.baseCoin.getFamily() === 'bch') {
+        const cashAddress = (wallet.baseCoin as Bch).canonicalAddress(output.address, 'cashaddr');
```

Comment:
are the addresses for `actualOutputs` always supposed to be in canonical form?

if so, please call `findMissingOutputs()` with `actualOutputs` mapped to the canonical address format

if you do that, the code changes for this method are not required

Reply:
it is not always be in cashaddr format, if user can pass base58 or cashaddr. So if user pass the base58 that will be picked up by the first if condition or for cashaddr we need to convert base58 to cashaddr and then check the condition.
Also we are updating expectedOutputs address with cashaddr so we can reuse for other function call.

Reply:
It is much simpler if we convert the addresses on both sides of the equation to the canonical format first.

Reply:
As far as I understand, `expectedOutputs` will always have `base58` addresses and `actualOutputs` will have either cashaddr or base58. If you pass `expectedOutputs` with addresses in canonical form this method is much simpler.

Reply:
got your point.  I updated the code as expectedOutputs can have cashaddr and base58 so we are converting into base58 for handling the flow.